pywr/recorders/recorders.py

```diff
@@ -110,7 +110,29 @@ def setup(self):
         if self.nodes is None:
             self.node_names = sorted(self.model.nodes.keys())
         else:
-            self.node_names = sorted(n.name for n in self.nodes)
+            # the next line is commented because when the nodes names are provided (e.g. loaded from the json)
+            # it will throw an exception
+            # self.node_names = sorted(n.name for n in self.nodes)
+            nodes = []
+            node_names = []
+            for node_ in self.nodes:
+                # test if the node name is provided
+                if isinstance(node_, basestring):
+                    try:
+                        # lookup node by name
+                        nodes.append(self.model.nodes[node_])
+                        node_names.append(node_)
+                    except:
+                        raise ValueError ("node name is not in the model: {}".format(node_))
+                else:
+                    try:
+                        nodes.append(node_)
+                        node_names.append((node_.name))
+                    except:
```

Comment:
except: without a specific subclass is too broad. This is only capturing an expected `AttributeError`? We can probably let that raise and drop the try-except?